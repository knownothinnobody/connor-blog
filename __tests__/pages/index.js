import React from "react"
import * as Gatsby from "gatsby"
import { render, shallow, mount } from "enzyme"

import Page from "../../src/pages/index"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

const cardNode = {
  frontmatter: {
    title: "CARD TITLE",
    posterImage: "example.com",
    date: "May 3, 2020",
  },
  fields: {
    slug: "/blog/test-slug",
  },
  timeToRead: 5,
  excerpt: "EXCERPT HERE",
}

describe("Index Page", () => {
  it("renders with 1 card correctly", () => {
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: "Site Title",
          description: "Site Description",
          author: "Author",
        },
      },
      allMarkdownRemark: {
        edges: [
          {
            node: cardNode,
          },
        ],
      },
    }))
    expect(shallow(<Page />)).toMatchSnapshot()
  })
})
