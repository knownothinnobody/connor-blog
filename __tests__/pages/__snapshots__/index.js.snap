// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders with 1 card correctly 1`] = `ShallowWrapper {}`;
